src/scaffoldmaker/meshtypes/meshtype_3d_cecum1.py

```diff
@@ -0,0 +1,1059 @@
+        endPoints_d2 = [innerEndPoints_d2, outerEndPoints_d2]
+
+        endDerivativesMap = [[None] * len(innerEndPoints_Id), [None] * len(innerEndPoints_Id)]
+        count = 0
+        for n1 in range(ei1Centre - ei1Left):
+            endDerivativesMap[0][count] = endDerivativesMap[1][count] = ((-1, 0, 0), (0, -1, 0), None)
+            count += 1
+
+        endDerivativesMap[0][count] = endDerivativesMap[1][count] = ((-1, 0, 0), (-1, -1, 0), None, (0, 1, 0))
+        count += 1
+        for n2 in range(ei2Top - ei2Bottom):
+            endDerivativesMap[0][count] = endDerivativesMap[1][count] = ((0, 1, 0), (-1, 0, 0), None)
+            count += 1
+        endDerivativesMap[0][count] = endDerivativesMap[1][count] = ((0, 1, 0), (-1, 1, 0), None, (1, 0, 0))
+        count += 1
+        for n1 in range(1, ei1Right - ei1Left + 1):
+            endDerivativesMap[0][count] = endDerivativesMap[1][count] = ((1, 0, 0), (0, 1, 0), None)
+            count += 1
+        endDerivativesMap[0][count] = endDerivativesMap[1][count] = ((1, 0, 0), (1, 1, 0), None, (0, -1, 0))
+        count += 1
+        for n2 in range(1, ei2Top - ei2Bottom + 1):
+            endDerivativesMap[0][count] = endDerivativesMap[1][count] = ((0, -1, 0), (1, 0, 0), None)
+            count += 1
+        endDerivativesMap[0][count] = endDerivativesMap[1][count] = ((0, -1, 0), (1, -1, 0), None, (-1, 0, 0))
+        count += 1
+        for n1 in range(ei1Right - ei1Centre):
+            endDerivativesMap[0][count] = endDerivativesMap[1][count] = ((-1, 0, 0), (0, -1, 0), None)
+            count += 1
+
+        ostiumSettings['Number of elements around ostium'] = len(innerEndPoints_Id)
+
+        nextNodeIdentifier, nextElementIdentifier, (o1_x, o1_d1, o1_d2, o1_d3, o1_NodeId, o1_Positions) = \
+            generateOstiumMesh(region, ostiumSettings, trackSurfaceOstium, centrePosition, axis1,
+                               nextNodeIdentifier, nextElementIdentifier)
+
+        fm = region.getFieldmodule()
+        mesh = fm.findMeshByDimension(3)
+        nodes = fm.findNodesetByFieldDomainType(Field.DOMAIN_TYPE_NODES)
+
+        startProportion = []
+        for n in range(len(innerEndPoints_Id)):
+            startProportion.append(trackSurfaceOstium.getProportion(o1_Positions[n]))
+
+        nextNodeIdentifier, nextElementIdentifier = createAnnulusMesh3d(
+            nodes, mesh, nextNodeIdentifier, nextElementIdentifier,
+            o1_x, o1_d1, o1_d2, None, o1_NodeId, None,
+            endPoints_x, endPoints_d1, endPoints_d2, None, endPoints_Id, endDerivativesMap,
+            elementsCountRadial = 2, tracksurface = trackSurfaceOstium, startProportion = startProportion,
+            endProportion = endProportion)
+
+        # Delete elements under annulus mesh
+        deleteElementsAndNodesUnderAnnulusMesh(fm, nodes, mesh, deleteElementIdentifier, deleteNodeIdentifier)
+
+        return annotationGroups
+
+    @classmethod
+    def generateMesh(cls, region, options):
```

Comment:
Remove so Scaffold_base.generateMesh() is used and implement @classmethod def refineMesh(cls, meshRefinement, options) which will be similar to this function but without any call to generateBaseMesh.
I suggest doing this for all your scaffolds once you see how it works.